File: tests/fixtures/sample.java
package com.example;

import java.util.List;

public class Service {
    public String name;
    private String secret;
    protected int value;

    public Service(String name) {
        this.name = name;
        this.secret = "";
        this.value = 0;
    }

    public String getName() {
        return this.name;
    }

    private void internal() {
        // hidden
    }

    protected void reset() {
        this.value = 0;
    }
}

public interface Reader {
    int read(byte[] buffer);
    void close();
}

public enum Color {
    RED,
    GREEN,
    BLUE;
}

public abstract class Base {
    public abstract void run();
}

File: tests/fixtures/sample.cs
using System;
using System.Collections.Generic;

namespace Example
{
    public class Service
    {
        public string Name { get; set; }
        private string _secret;
        protected int Value { get; set; }

        public Service(string name)
        {
            Name = name;
            _secret = "";
        }

        public string GetName()
        {
            return Name;
        }

        private void Internal()
        {
            // hidden
        }
    }

    public interface IReader
    {
        int Read(byte[] buffer);
        void Close();
    }

    public enum Color
    {
        Red,
        Green,
        Blue
    }

    public struct Point
    {
        public int X;
        public int Y;
    }

    public record Person(string Name, int Age);

    public abstract class Base
    {
        public abstract void Run();
    }
}
